File: 프로그래머스/구명보트/savingboat.cpp
    #include<iostream>
    #include <string>
    #include <vector>
    #include <algorithm>
    #include <deque>
    using namespace std;

    int solution(vector<int> people, int limit) {
        int answer = 0;
        sort(people.begin(), people.end(),greater<int>());
        int start = 0;
        int end = people.size()-1;
        for (int i = start; i <=end; i++) {
            int sum = people[i];
            int j = end;
            for (j; j >= start; j--) {
                if (sum + people[j] <= limit) {
                    sum += people[j];
                    end -= 1;
                }
                else {
                    break;
                }
            }
            start += 1;
            answer += 1;
        }
        return answer;
    }
    int main() {
        int start = 0;
        int end = 5;
        int a = solution({ 70,50,80,50 }, 100);
        cout << a;
    }

File: 프로그래머스/구명보트/savingboat.java
import java.util.*;
class Solution {
    public int solution(int[] people, int limit) {
        int answer = 0;
        int leftsize =0;
        int rightsize = people.length-1;
        Arrays.sort(people);
        while(leftsize<=rightsize){
            int curlimit = limit;
            curlimit -= people[rightsize];
            rightsize--;
            if(curlimit >= people[leftsize])
                leftsize++;
            answer++;
        }
        return answer;
    }
}
public class test{
    public static void main(String[] args){
        Solution a = new Solution();
        int z = a.solution(new int[]{70,80,50},100);
        System.out.println(z);
    }
}
